docs/source/administrator/upgrading/upgrade-1-to-2.md

```diff
@@ -0,0 +1,141 @@
+# Major upgrade: 1.\* to 2.\*
+
+Z2JH 2 contains several breaking changes, including some that affect the security of your deployment.
+This guide will help you upgrade from 1.\* to 2.\*.
+
+## Security: breaking change to `*.networkPolicy.egress`
+
+NetworkPolicy egress rules have been extended with a new property.
+If you have configured any of:
+
+- `hub.networkPolicy.egress`
+- `proxy.chp.networkPolicy.egress`
+- `proxy.traefik.networkPolicy.egress`
+- `singleuser.networkPolicy.egress`
+
+you must review your configuration as additional default egress routes have been added.
+Previously `*.networkPolicy.egress` controlled all egress but a new properties `*.networkPolicy.egressAllowRules` add additional egress rules by default.
+
+If you have configured `*.networkPolicy.egress` for `hub`, `proxy.chp`,
+`proxy.traefik` or `singleuser` to restrict the permissions to establish
+outbound network connections, then this upgrade is likely to _escalate those
+permissions unless you revise your configuration_. The new configuration
+`*.networkPolicy.egressAllowRules` are by default granting most of the egress
+permissions previously granted by default via the `*.networkPolicy.egress`
+configuration, and `*.networkPolicy.egress` are now by default not providing
+any permissions.
+
+If you for example had overridden the previously very permissive default value
+of `singleuser.networkPolicy.egress` to be less permissive, you should consider
+disabling all `singleuser.networkPolicy.egressAllowRules` like this
+to not risk escalating the permissions.
+
+```yaml
+singleuser:
+  networkPolicy:
+    egressAllowRules:
+      cloudMetadataServer: false
+      dnsPortsPrivateIPs: false
+      nonPrivateIPs: false
+      privateIPs: false
+```
+
+For more details, see the documentation on [Kubernetes Network Policies](netpol)
+and the configuration reference entries under
+[`*.networkPolicy.egress`](schema_hub.networkPolicy.egress) and
+[`*.networkPolicy.egressAllowRules`](schema_hub.networkPolicy.egressAllowRules).
+
+## JupyterHub 2 and related hub components
+
+Z2JH 2.0.0 upgrades JupyterHub to the 2.\* series, and also upgrades all hub components.
+If you are using any custom JupyterHub services, addons, API integrations, or extra configuration, you should review the breaking changes in the
+[JupyterHub 2.0.0 changelog](https://github.com/jupyterhub/jupyterhub/blob/2.3.1/docs/source/changelog.md#200).
+
+JupyterHub 2 uses an updated database schema.
+Z2JH 2.0.0 automatically handles the upgrade for SQLite databases (the default), but it will not be possible to downgrade to older releases after this.
```

Comment:
```suggestion
Z2JH 2.0.0 automatically handles the upgrade for SQLite databases (the default), but if you use an external database you need to configure [`hub.db.upgrade`](schema_hub.db.upgrade) to true when upgrading. It will not be possible to downgrade to older releases after this without also using a backup or resetting the database.
```